third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -866,10 +919,18 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
-	// Update only updates the settings, so they are all we need to set.
-	instance := &sqladmin.DatabaseInstance{
-		Settings: expandSqlDatabaseInstanceSettings(d.Get("settings").([]interface{}), !isFirstGen(d)),
+	instance := &sqladmin.DatabaseInstance{}
+	<% unless version == 'ga' -%>
+	if sqlDatabaseIsOnPremises(d) {
```

Comment:
same comment re: if/elses. I'd rather just set things and allow the client library to filter out nils. You can add an expander for OnPremisesConfiguration that'll return nil if it isn't set, similarly to how we do for settings.